app/controllers/MovementsController.scala

```diff
@@ -20,87 +20,166 @@ import java.time.OffsetDateTime
   cc: ControllerComponents,
   arrivalMovementRepository: ArrivalMovementRepository,
   arrivalMovementService: ArrivalMovementService,
+  authenticatedOptionalArrival: AuthenticatedGetOptionalArrivalForWriteActionProvider,
   authenticate: AuthenticateActionProvider,
+  authenticateForWrite: AuthenticatedGetArrivalForWriteActionProvider,
   messageConnector: MessageConnector
 )(implicit ec: ExecutionContext)
     extends BackendController(cc) {
 
   private val logger = Logger(getClass)
 
-  def createMovement: Action[NodeSeq] = authenticate().async(parse.xml) {
+  def createMovement: Action[NodeSeq] = authenticatedOptionalArrival().async(parse.xml) {
     implicit request =>
-      arrivalMovementService.makeArrivalMovement(request.eoriNumber)(request.body) match {
+      request.arrival match {
+        case Some(arrival) => update_internal(arrival, request.body)
         case None =>
-          Future.successful(BadRequest("Invalid data: missing either DatOfPreMES9, TimOfPreMES10 or DocNumHEA5"))
-
-        case Some(x) =>
-          x.flatMap {
-              arrival =>
-                arrivalMovementRepository.insert(arrival) flatMap {
-                  _ =>
-                    messageConnector
-                      .post(TransitWrapper(request.body), MessageType.ArrivalNotification, OffsetDateTime.now)
-                      .flatMap {
-                        _ =>
-                          val newState = arrival.state.transition(SubmissionResult.Success)
-                          arrivalMovementRepository
-                            .setState(arrival.arrivalId, newState)
-                            .map {
-                              _ =>
-                                Accepted("Message accepted")
-                                // TODO: This needs to be replaced url to arrival movement resource, for which we need an Arrival Movement number
-                                  .withHeaders("Location" -> arrival.arrivalId.index.toString)
-
-                            }
-                            .recoverWith {
-                              case e: Exception =>
-                                logger.error(s"setState failed with following Exception: ${e.getMessage}")
-                                Future.successful(InternalServerError)
-                            }
-                      }
-                      .recoverWith {
-                        case error: Exception =>
-                          logger.error(s"Call to EIS failed with the following Exception: ${error.getMessage}")
-                          val newState = arrival.state.transition(SubmissionResult.Failure)
-                          arrivalMovementRepository.setState(arrival.arrivalId, newState).map {
+          arrivalMovementService.makeArrivalMovement(request.eoriNumber)(request.body) match {
+            case None =>
+              Future.successful(BadRequest("Invalid data: missing either DatOfPreMES9, TimOfPreMES10 or DocNumHEA5"))
```

Comment:
Maybe something for future consideration, but is the API going to offer custom error codes - probably not if we're not returning JSON to the consumer of the API but worth thinking about)